downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/DatabaseDumps.java

```diff
@@ -0,0 +1,79 @@
+		// Take mysqldumps of 'stable_identifiers' and 'test_reactome_##' and compress them using gzip.
+		
+		//TODO: The gzipped and non-gzipped files might be getting moved 
+		
+		logger.info("Generating DatabaseDumps");
+		// Create databases folder that will hold the DB dumps
+		File databaseDir = new File(releaseNumber + "/databases");
+		if (!databaseDir.exists()) {
+			databaseDir.mkdir();
+		}
+		// Start with stable_identifiers dump
+		//TODO: Sticking to GK naming conventions?
+		logger.info("Dumping stable_identifiers to gk_stable_ids.sql...");
+		File stableIdsFile = new File(releaseNumber + "/databases/gk_stable_ids.sql");
+		String[] stableIdsCommand = new String[]{"mysqldump", "-h" + host,"-u" + username, "-p" + password, "-P" + port, "stable_identifiers"};
+		ProcessBuilder stableIdsprocessBuilder = new ProcessBuilder(Arrays.asList(stableIdsCommand));
+		stableIdsprocessBuilder.redirectError(Redirect.INHERIT);
+		stableIdsprocessBuilder.redirectOutput(Redirect.to(stableIdsFile));
+		Process stableIdsProcess = stableIdsprocessBuilder.start();
+		stableIdsProcess.waitFor();
+		
+		byte[] buffer = new byte[1024];
+		GZIPOutputStream gzosStableIds = new GZIPOutputStream(new FileOutputStream(releaseNumber + "/databases/gk_stable_ids.sql.gz"));
+		FileInputStream fisStableIds = new FileInputStream(releaseNumber + "/databases/gk_stable_ids.sql");
+		int lengthStId;
+		while ((lengthStId = fisStableIds.read(buffer)) > 0) {
+			gzosStableIds.write(buffer, 0, lengthStId);
+		}
+		fisStableIds.close();
+		gzosStableIds.finish();
+		gzosStableIds.close();
+		Files.delete(Paths.get(releaseNumber + "/databases/gk_stable_ids.sql"));
+		// Now test_reactome dump
+		logger.info("Dumping test_reactome_" + releaseNumber + " to gk_current.sql");
+		File gkCurrentFile = new File(releaseNumber + "/databases/gk_current.sql");
+		
+		String[] gkCurrentCommand = new String[]{"mysqldump", "-h" + host,"-u" + username, "-p" + password, "-P" + port, testReactomeDatabase};
+		ProcessBuilder gkCurrentProcessBuilder = new ProcessBuilder(Arrays.asList(gkCurrentCommand));
+		gkCurrentProcessBuilder.redirectError(Redirect.INHERIT);
+		gkCurrentProcessBuilder.redirectOutput(Redirect.to(gkCurrentFile));
+		
+		Process gkCurrentProcess = gkCurrentProcessBuilder.start();
+		gkCurrentProcess.waitFor();
+		
+		GZIPOutputStream gzosGkCurrent = new GZIPOutputStream(new FileOutputStream(releaseNumber + "/databases/gk_current.sql.gz"));
+		FileInputStream fisGkCurrent = new FileInputStream(releaseNumber + "/databases/gk_current.sql");
+		int lengthGk;
+		while ((lengthGk = fisGkCurrent.read(buffer)) > 0) {
+			gzosGkCurrent.write(buffer, 0, lengthGk);
+		}
+		fisGkCurrent.close();
+		gzosGkCurrent.finish();
+		gzosGkCurrent.close();
+		
+		Files.delete(Paths.get(releaseNumber + "/databases/gk_current.sql"));
```

Comment:
Creation of stable_identifiers dump and test_reactome dump share a lot in common.  Could the common code be extracted to a method called once for each?

Reply:
That would probably be a larger refactoring, maybe better saved for the next release of download-directory.

Reply:
Sure, not mandatory right now.  I'm on a refactoring kick lately and try to be aggressive with aggregating common code to one place.